### ISSUE TITLE
alternate way to represent uninitialized string

### ISSUE THREAD

**Issue by jlapeyre** [davidanthoff/StringBuilders.jl#9]
Hi. At JuliaCon, I mentioned using `Union{String, Nothing}`. But, in fact, I mis-remembered how I upgraded to v0.7. In the end, I used `as_string::String` and simply initialized `as_string` to `""`.

I see you have used `Union{String, Nothing}`.  [Here](https://github.com/jlapeyre/StringBuilders.jl/blob/gjl/upgradev0.7/src/StringBuilders.jl) is my implementation. I can't see any compelling reason to prefer one solution over the other. But, I thought it worth mentioning the alternative.

I also included this test
```julia
@testset "initialization" begin
    sb = StringBuilder()
    @test isempty(String(sb))
end
```
Both implementations pass this test.

**Comment by davidanthoff** [comment]
The ``Union`` implementation seems a bit cleaner? But would be good to add that test, could you open a PR?

**Comment by jlapeyre** [comment]
I prefer the `String` implementation. But, that's probably because I wrote it, not because I have a good argument. I tried the `Union` implementation first, and then switched to `String`. In it's favor, you might argue that `Union{String, Nothing}` signals clearly in the code when nothing has been written to the buffer. I say, leave it as you have it. If this ever becomes widely used, then maybe benchmarking would be worthwhile.
